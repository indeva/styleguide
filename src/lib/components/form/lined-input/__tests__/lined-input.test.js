import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import LinedInput from '..';

describe('LinedInput', () => {
  const defaultProps = {
    fieldRef: jest.fn(),
    id: 'name',
    name: 'name',
    value: 'value',
    label: 'Name',
    type: 'text',
    hasError: false,
    onChange: () => {},
  };

  describe('snapshot tests', () => {
    describe('when the component has just the defaultprops', () => {
      it('renders the LinedInput normally', () => {
        const component = renderer.create(<LinedInput {...defaultProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when the component also has the flag hasError as prop', () => {
      it('renders the LinedInput with error', () => {
        const component = renderer.create(
          <LinedInput {...defaultProps} hasError />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('behavior tests', () => {
    describe('when the value has been changed', () => {
      it('calls the callback function', () => {
        const onChangeCallback = jest.fn();
        const wrapper = mount(
          <LinedInput
            {...defaultProps}
            onChange={onChangeCallback}
          />,
        );

        wrapper.find('input').simulate('change');
        expect(onChangeCallback).toHaveBeenCalledTimes(1);
      });
    });

    describe('when the field has key pressed', () => {
      it('calls the callback function', () => {
        const onKeyDown = jest.fn();
        const wrapper = mount(
          <LinedInput
            {...defaultProps}
            onKeyDown={onKeyDown}
          />,
        );

        wrapper.find('input').simulate('keyDown');
        expect(onKeyDown).toHaveBeenCalledTimes(1);
      });
    });
  });
});
