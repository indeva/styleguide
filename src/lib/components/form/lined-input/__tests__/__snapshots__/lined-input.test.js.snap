// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinedInput snapshot tests when the component has just the defaultprops renders the LinedInput normally 1`] = `
<div
  className="lined-input"
>
  <input
    className="lined-input__field"
    disabled={false}
    id="name"
    name="name"
    onChange={[MockFunction]}
    required={true}
    type="text"
    value="value"
  />
  <label
    className="lined-input__label"
    htmlFor="name"
  >
    Name
  </label>
  <div />
</div>
`;

exports[`LinedInput snapshot tests when the componente also has the flag hasError as prop renders the LinedInput with error 1`] = `
<div
  className="lined-input"
>
  <input
    className="lined-input__field lined-input__field--error"
    disabled={false}
    id="name"
    name="name"
    onChange={[MockFunction]}
    required={true}
    type="text"
    value="value"
  />
  <label
    className="lined-input__label lined-input__label--error"
    htmlFor="name"
  >
    Name
  </label>
  <div />
</div>
`;
